### DISCUSSION TITLE
charon-cmd support for childless SA

### DISCUSSION THREAD

**Discussion by psalty** [strongswan/strongswan#1594] [Ideas]
Hi. I think it is would be a good feature if we have childless sa support in the road-warrior cases.

Because whoever initiates childless sa is mostly going to be the client.

`swanctl -i --ike conn_name`

` charon-cmd --rfc6023 --host name --identity `

Is it hard to implement?

If I want to make it on my own for testing, where is the place to look into?

Thanks.

**Comment by tobiasbrunner** [comment]
You want to only establish an IKE_SA (i.e. no CHILD_SA at all)? To what end?

**Reply by psalty** [reply]
StrongSwan server

**Reply by tobiasbrunner** [reply]
What do you mean?

**Reply by psalty** [reply]
charon-cmd to strongswan server for ike auth only purpose. Not necessarily against any other VPN GW.

**Reply by tobiasbrunner** [reply]
But for what purpose?

**Reply by psalty** [reply]
authentication only. make encryption optional.

**Reply by tobiasbrunner** [reply]
I don't get what you mean.

**Reply by psalty** [reply]
I want to charon-cmd initiate IKE only (childless SA just like swanctl -i --ike does) to the strongswan peer server.
Technically I want to do the exact same thing what swanctl -i --ike does except I don't want to run any charon service in the background as a client.
I don't know if I explained well enough.

**Reply by tobiasbrunner** [reply]
I get WHAT you want to do, but I don't get WHY you want to do it. Especially with `charon-cmd` as a childless IKE_SA serves absolutely no purpose there.

**Reply by psalty** [reply]
Is charon-cmd for server-side at all? I thought that is for the client side only, and that is my context.
Why?
Don't want encryption until I want it, but want to make the client authenticated to the server.

**Reply by tobiasbrunner** [reply]
> Is charon-cmd for server-side at all? I thought that is for the client side only, and that is my context.

It's not, but what triggered that question?

> Don't want encryption until I want it, but want to make the client authenticated to the server.

Authenticated for what? If you don't protect traffic with IPsec there is absolutely no reason to authenticate the client (note that encryption is optional with ESP, you could also just authenticate and integrity protect the traffic if encryption is an issue for some reason). Also, with `charon-cmd` you can't add encryption later without restarting the program (unlike with the daemon where you could theoretically configure and negotiate a CHILD_SA later on).

**Reply by psalty** [reply]
I think I want to use Charon-cmd only to be able to maintain IKE SA on the trusted network.

No intention of later encryption to be clear.

**Reply by tobiasbrunner** [reply]
That's completely useless.

**Reply by psalty** [reply]
per RFC6023 usage cases

o  Location-aware security, as in [[SecureBeacon](https://www.rfc-editor.org/rfc/rfc6023#ref-SecureBeacon)].  The user is
      roaming between trusted and untrusted networks.  While in an
      untrusted network, all traffic should be encrypted, but on the
      trusted network, only the IKEv2 SA needs to be maintained.

**Reply by tobiasbrunner** [reply]
You might have misunderstood that. Without the "roaming between trusted and untrusted networks" part such an IKE_SA is completely useless.

**Reply by psalty** [reply]
Thanks.

**Comment by corsac-s** [comment]
I think the main point of RFC 9073 is not to have perpetually childless IKE SA, but rather making sure the CHILD SA are created with different key material than the IKE SA. I had the feeling the CHILD SA was to be created automatically as soon as some real traffic happens (in one direction or another) using CREATE_CHILD_SA but I might be wrong.

In any case, it'd be nice to support that in all daemons.

**Reply by tobiasbrunner** [reply]
Adding a `--childless [force]` argument would be an option for that use case. A Child SA would still get initiated but we'd either prefer (if supported by the responder) or force creating it with a separate CREATE_CHILD_SA exchange after establishing a childless IKE SA. I've pushed something to that effect to the `1594-cmd-childless` branch.
